samples/astarte_app/main/src/astarte_task.c

```diff
@@ -97,3 +225,103 @@ void astarte_task_entry(void *ctx)
  * Static functions definitions
  ***********************************************/
+
+static void astarte_task_transmit(void *ctx)
+{
+    vTaskSuspend(NULL);
+    astarte_device_handle_t device = *((astarte_device_handle_t *) ctx);
+
+#if defined(CONFIG_DEVICE_INDIVIDUAL_TRANSMISSION)
+    ESP_LOGI(TAG, "Waiting %d seconds to send individuals.",
+        CONFIG_DEVICE_INDIVIDUAL_TRANSMISSION_DELAY_SECONDS);
+    vTaskDelay(
+        CONFIG_DEVICE_INDIVIDUAL_TRANSMISSION_DELAY_SECONDS * MS_IN_SEC / portTICK_PERIOD_MS);
+    sample_individual_transmission(device);
+#endif
+#if defined(CONFIG_DEVICE_OBJECT_TRANSMISSION)
+    ESP_LOGI(TAG, "Waiting %d seconds to send objects.",
+        CONFIG_DEVICE_OBJECT_TRANSMISSION_DELAY_SECONDS);
+    vTaskDelay(CONFIG_DEVICE_OBJECT_TRANSMISSION_DELAY_SECONDS * MS_IN_SEC / portTICK_PERIOD_MS);
+    sample_object_transmission(device);
+#endif
+#if defined(CONFIG_DEVICE_PROPERTY_SET_TRANSMISSION)
+    ESP_LOGI(TAG, "Waiting %d seconds to set properties.",
+        CONFIG_DEVICE_PROPERTY_SET_TRANSMISSION_DELAY_SECONDS);
+    vTaskDelay(
+        CONFIG_DEVICE_PROPERTY_SET_TRANSMISSION_DELAY_SECONDS * MS_IN_SEC / portTICK_PERIOD_MS);
+    sample_property_set_transmission(device);
+#endif
+#if defined(CONFIG_DEVICE_PROPERTY_UNSET_TRANSMISSION)
+    ESP_LOGI(TAG, "Waiting %d seconds to unset properties.",
+        CONFIG_DEVICE_PROPERTY_UNSET_TRANSMISSION_DELAY_SECONDS);
+    vTaskDelay(
+        CONFIG_DEVICE_PROPERTY_UNSET_TRANSMISSION_DELAY_SECONDS * MS_IN_SEC / portTICK_PERIOD_MS);
+    sample_property_unset_transmission(device);
+#endif
+
+#if !defined(CONFIG_DEVICE_INDIVIDUAL_TRANSMISSION) && !defined(CONFIG_DEVICE_OBJECT_TRANSMISSION) \
+    && !defined(CONFIG_DEVICE_PROPERTY_SET_TRANSMISSION)                                           \
+    && !defined(CONFIG_DEVICE_PROPERTY_UNSET_TRANSMISSION)
+    ESP_LOGI(TAG, "No trasmission required, waiting for operational timeout.");
+    vTaskDelay(CONFIG_DEVICE_OPERATIONAL_TIMEOUT * MS_IN_SEC / portTICK_PERIOD_MS);
+#endif
+
+    vTaskDelay(MS_IN_SEC / portTICK_PERIOD_MS);
+    ESP_LOGI(TAG, "Transmission task completed signaling the main task to terminate.");
+    xTaskNotifyGive(device_task_handle);
+    vTaskDelete(NULL);
+}
+
+static void connection_callback(astarte_device_connection_event_t event)
+{
+    (void) event;
+    ESP_LOGI(TAG, "Astarte device connected.");
+    vTaskResume(transmit_task_handle);
+}
+
+static void disconnection_callback(astarte_device_disconnection_event_t event)
+{
+    (void) event;
+    ESP_LOGI(TAG, "Astarte device disconnected");
+}
+
+static void datastream_individual_callback(astarte_device_datastream_individual_event_t event)
+{
+    const char *interface_name = event.base_event.interface_name;
+    const char *path = event.base_event.path;
+    astarte_data_t individual = event.data;
+
+    ESP_LOGI(TAG, "Datastream individual event, interface: %s, path: %s", interface_name, path);
```

Comment:
Do we want to keep log level to info during the API restructuring? Otherwise I would like to use Debug as level when SDK transmits data to Astarte. 
```suggestion
    ESP_LOGD(TAG, "Datastream individual event, interface: %s, path: %s", interface_name, path);
```

Reply:
Since this is an example, I would leave the level to info. This shows the user of the sample that a data-stream has been received.